### ISSUE TITLE
Optional bug

### ISSUE THREAD

**Issue by tamuhey** [microsoft/pyright#1425]
```python
def f(a: Optional[str]):
    a.startswith("foo") # Should error, but Pyright accepts
```

Pyright should output error for the above code, becuase `a` is a `None | str`.

**Comment by erictraut** [comment]
If you want pyright to report an error in this situation, you need to enable the `reportOptionalMemberAccess` diagnostic check. It's off by default in "basic" type checking mode and on by default in "strict".

```python
# pyright: reportOptionalMemberAccess=true

from typing import Optional

def f(a: Optional[str]):
    a.startswith("foo") # Error reported
```
